Formula/mysql-connector-c++.rb

```diff
@@ -1,9 +1,3 @@
-class MysqlClientRequirement < Requirement
-  fatal true
-  default_formula "mysql"
-  satisfy { which "mysql" }
```

Comment:
This is not quite equivalent as after this PR it'll now check for `mysql_config` instead. I'm fine with that but just so you're aware.